src/Microsoft.Health.Fhir.Core/Features/Conformance/Serialization/DefaultOptionHashSetJsonConverter.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance.Serialization
+{
+    internal class DefaultOptionHashSetJsonConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            EnsureArg.IsNotNull(writer, nameof(writer));
+            EnsureArg.IsNotNull(value, nameof(value));
+            EnsureArg.IsNotNull(serializer, nameof(serializer));
+
+            dynamic defaultOption = ((dynamic)value).DefaultOption;
```

Comment:
Instead of using dynamic, could define an interface like IDefaultOption or something which expose a property `object DefaultOption` then cast the object to that interface and access the property? It guards against potential renaming in the future and also no performance hit?